image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -0,0 +1,108 @@
+  In Indexer mode, the Scanner pulls images and analyzes them to determine the base OS and installed packages
+  (i.e. it indexes the images).
+  In Matcher mode, the Scanner matches the found packages to known vulnerabilities to produce a Vulnerability Report.
+  Both modes require access to a PostgreSQL database.
+
+  StackRox's Central service has two Scanner deployments: a Scanner running in Indexer mode and another
+  running in Matcher mode. In this context, the Helm chart can create its own certificates.
+
+  StackRox's Secured Cluster services may deploy the Scanner in Indexer mode, only.
+  This would be done to access registries inaccessible to the Central cluster.
+  In this context, the Helm chart does not generate its own certificates.
+
+  $scannerV4Config contains all values which are configured by the user. The structures can be viewed in the respective
+  config-shape. See internal/scanner-v4-config-shape.yaml.
+   */}}
+
+{{ define "srox.scannerV4Init" }}
+
+{{ $ := index . 0 }}
+{{ $scannerV4Cfg := index . 1 }}
+{{ $_ := false }}
+
+{{/* Handle high-level defaulting logic. */}}
+{{/* Treat a scannerV4.disable=false setting as a hint to enable indexer and/or matcher depending on the chart being used. */}}
+{{- if and (not $scannerV4Cfg.disable) (kindIs "invalid" $scannerV4Cfg.indexer.disable) (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
+{{- /* The top-level setting scannerV4.disable=false activates a default setting based on the chart. */ -}}
+{{- if eq $.Chart.Name "stackrox-central-services" -}}
+  {{- $_ = set $scannerV4Cfg.indexer "disable" false -}}
+  {{- $_ = set $scannerV4Cfg.matcher "disable" false -}}
+{{- else if eq $.Chart.Name "stackrox-secured-cluster-services" -}}
+  {{- $_ = set $scannerV4Cfg.indexer "disable" false -}}
+{{- else -}}
+  {{- include "srox.fail" (printf "Unexpected Helm chart name %q." $.Chart.Name) -}}
+{{- end -}}
+{{- else -}}
+{{- end -}}
+
+{{/* Sanity checks. */}}
+{{- if eq $.Chart.Name "stackrox-central-services" -}}
+  {{- if and (not $scannerV4Cfg.indexer.disable) $scannerV4Cfg.matcher.disable -}}
+    {{- include "srox.fail" "This chart only supports deploying indexer and matcher together." -}}
+  {{- end -}}
+  {{- if and (not $scannerV4Cfg.matcher.disable) $scannerV4Cfg.indexer.disable -}}
+    {{- include "srox.fail" "This chart only supports deploying indexer and matcher together." -}}
+  {{- end -}}
+{{- end -}}
+{{- if eq $.Chart.Name "stackrox-secured-cluster-services" -}}
+  {{- if not $scannerV4Cfg.matcher.disable -}}
+    {{- include "srox.fail" "This chart only supports deploying the scanner V4 indexer." -}}
+  {{- end -}}
+{{- end -}}
+
+{{/* Provide some human-readable feedback regarding the Scanner V4 configuration to the user installing the Helm chart. */}}
+{{- $scannerV4Mode := "none" -}}
+{{- if and (not $scannerV4Cfg.indexer.disable) (not $._rox.scannerV4.matcher.disable) -}}
+{{- $scannerV4Mode = "indexer-and-matcher" -}} 
```

Comment:
> What we, on the other hand, leverage here is the ability to spawn two distinct scanner V4 instances, where one has the indexer hat on and the other the matcher hat. So, it is strictly speaking, something different, which is why I came up with this distinct terminology above.

You are right. If running two different deployments, one for the Indexer and another for the Matcher, we are not enabling "combo mode". I got confused by the name of the variable, "ScannerV4Mode". This "mode" is not the same "mode" I was referring to. If the variable was named `scannerV4InstallationMethod` and did not reference the word "mode" it would make it a little bit more clear.

I added a comment to change the customer-facing label for this.